output-validation-rule.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var assert = require('assert');
+var Joi = require('joi');
+
+module.exports = OutputValidationRule;
+
+function OutputValidationRule(status, spec) {
+  assert(status, 'OutputValidationRule: missing status param');
+  assert(spec, 'OutputValidationRule: missing spec param');
+
+  this.ranges = status.split(',').map(trim).filter(Boolean).map(rangify);
+  this.status = status;
+  this.spec = spec;
+
+  this.validateInput();
+}
+
+/**
+ * Validates it's input values
+ *
+ * @throws Error
+ */
+
+OutputValidationRule.prototype.validateInput = function validateInput() {
+  assert(/^[0-9,\-]+|\*$/.test(this.status),
```

Comment:
The regex needs parens e.g. `/^(A|B)$/` otherwise the `|` takes precedence e.g. /^A/ or /B$/, so '100x' would pass the original regex. Also, this doesn't check for other invalid cases like '100--200', '100-', '-'.